source/operators/captions/webcaptions.py

```diff
@@ -0,0 +1,988 @@
+                self.operator_name_with_lang = self.operator_object.name+"_"+self.source_language_code
+            if "TargetLanguageCode" in self.operator_object.configuration:
+                self.target_language_code = self.operator_object.configuration["TargetLanguageCode"]
+            if "ExistingSubtitlesObject" in self.operator_object.configuration:
+                self.existing_subtitles_object = self.operator_object.configuration["ExistingSubtitlesObject"]
+                self.existing_subtitles = True
+        except KeyError as e:
+            self.operator_object.update_workflow_status("Error")
+            self.operator_object.add_workflow_metadata(WebCaptionsError="No valid inputs {e}".format(e=e))
+            raise MasExecutionError(operator_object.return_output_object())
+
+    def WebCaptionsOperatorName(self, language_code=None, source=""):
+
+        # Shouldn't assume WebCaptions operator is WebCaptions, maybe pass it in the configuration?
+        print("WebCaptionsOperatorName {}, {}".format(language_code, source))
+
+        operator_name = "WebCaptions"+source
+
+        if language_code != None:
+            return operator_name+"_"+language_code
+
+        try:
+            name = operator_name+"_"+self.source_language_code
+        except KeyError:
+            self.operator_object.update_workflow_status("Error")
+            self.operator_object.add_workflow_metadata(WebCaptionsError="Missing language code for WebCaptions {e}".format(e=e))
+            raise MasExecutionError(self.operator_object.return_output_object())
+
+        print("WebCaptionsOperatorName() Name {}".format(name))
+        return name
+
+    def CaptionsOperatorName(self, language_code=None):
+
+        # Shouldn't assume WebCaptions operator is WebCaptions, maybe pass it in the configuration?
+        operator_name = "Captions"
+
+        if language_code != None:
+            return operator_name+"_"+language_code
+
+        try:
+            name = operator_name+"_"+self.source_language_code
+        except KeyError:
+            self.operator_object.update_workflow_status("Error")
+            self.operator_object.add_workflow_metadata(WebCaptionsError="Missing language code for WebCaptions {e}".format(e=e))
+            raise MasExecutionError(self.operator_object.return_output_object())
+
+        print("CaptionsOperatorName() Name {}".format(name))
+        return name
+
+
+
+    def GetTranscript(self):
+
+        transcript = []
+
+        response = dataplane.retrieve_asset_metadata(self.asset_id, operator_name=self.transcribe_operator_name)
+        transcript.append(response["results"])
+
+        while "cursor" in response:
+            response = dataplane.retrieve_asset_metadata(self.asset_id, operator_name=self.transcribe_operator_name, cursor=response["cursor"])
+            transcript.append(response["results"])
+
+        return transcript
```

Comment:
Just a general comment, this is a good example of doing a paginated GET from the dataplane